nubia/internal/cmdbase.py

```diff
@@ -53,7 +59,7 @@ def built_in(self) -> bool:
     def set_command_registry(self, command_registry):
         self._command_registry = command_registry
 
-    def run_interactive(self, cmd, args, raw):
+    async def run_interactive(self, cmd, args, raw):
```

Comment:
@AhmedSoliman so we want to keep backwards compatibility, but at the same time allow for this to be async. I think that ideally we would make this visible in the typing, but nubia isn't typed (yet?). So making it visibly `async` here would signify that it can be async, but should still allow for sync as well, right? So I think making these `async` is the right decision?

@SelleslaghGianni `run_cli`, `add_arguments`, and perhaps some others should also be made `async` then, as they are in `AutoCommand` below.